src/expression.d

```diff
@@ -3556,19 +3556,12 @@ public:
                     }
                 }
                 s = s.toAlias();
+
                 // Same as wthis.ident
-                if (s.needThis() || s.isTemplateDeclaration())
-                {
-                    e = new VarExp(loc, withsym.withstate.wthis);
-                    e = new DotIdExp(loc, e, ident);
-                }
-                else
-                {
-                    Type t = withsym.withstate.wthis.type;
-                    if (t.ty == Tpointer)
-                        t = (cast(TypePointer)t).next;
-                    e = typeDotIdExp(loc, t, ident);
```

Comment:
In `WithStatement.semantic`, if the `exp` operand is a runtime expression, it's assigned to the temporary variable `wthis`. Therefore, i `wthis` is not null in here, it should be converted to a `VarExp` always.